jar-infer/jar-infer-lib/src/test/java/com/uber/nullaway/jarinfer/JarInferTest.java

```diff
@@ -507,6 +499,36 @@ public void testSignedJars() throws Exception {
         EntriesComparator.compareManifestContents(outputJarPath, baseJarPath));
   }
 
+  /** copy the jar at {@code baseJarPath} to a signed jar at {@code signedJarPath} */
+  private void copyAndSignJar(String baseJarPath, String signedJarPath)
+      throws IOException, InterruptedException {
+    Files.copy(
+        Paths.get(baseJarPath), Paths.get(signedJarPath), StandardCopyOption.REPLACE_EXISTING);
+    String ksPath =
+        Thread.currentThread().getContextClassLoader().getResource("testKeyStore.jks").getPath();
+    // A public API for signing jars was added for Java 9+, but there is only an internal API on
+    // Java 8.  And we need the test code to compile on Java 8.  For simplicity and uniformity, we
+    // just run jarsigner as an executable (which slightly slows down test execution)
+    String jarsignerExecutable =
+        String.join(
+            FileSystems.getDefault().getSeparator(),
+            new String[] {System.getenv("JAVA_HOME"), "bin", "jarsigner"});
```

Comment:
Just to double check: we already depend on `JAVA_HOME` being set to run/test NullAway?

Edit: I normally wouldn't want a unit test depending on an env var, but it looks like `gradlew` already sets/checks `JAVA_HOME`, so this should be fine.

Reply:
Yes, I was a bit hesitant here but since Gradle already relies on `JAVA_HOME` I thought it was ok. The alternative would be a bunch of gross reflection to call the right APIs on Java 11 while making the code still compile on Java 8, which seemed worse.